frontend/src/components/Block/Block.tsx

```diff
@@ -106,6 +94,7 @@ const Block = () => {
     const onNext = async (doBreak = false) => {
         if (!doBreak && actions.length) {
             updateActions(actions);
+            setCurrentAction(currentActionIndex! + 1); // Increment current action index
```

Comment:
There shouldn't be any problems with this whenever a new round occurs since we set the index back to `0` when that happens.